File: config/filetemplates/Java/FieldValidation.java
#if (${PACKAGE_NAME} != "")package ${PACKAGE_NAME};
#end
import com.tplus.transform.runtime.*;
import com.tplus.transform.runtime.handler.*;

public class ${NAME} implements IFieldValidation {

    public void validate(String fldName, Object value, DataObject obj) throws FieldValidationException {		
        //check value here
        //if(failed)
        //throw new FieldValidationException("CostValidation: Cost for book '"+ bookSec.getName() +"' should be less than 6000 for subject '" + subSec.getName() + "'", "cost");		
    }
}

File: config/filetemplates/CPP/FieldValidation.cpp
\#include "Transform.h"

class ${NAME} : public IFieldValidation{
    public:	bool validate(const vstring& fldName, const Object* value, DataObject* obj) {
       //check value here
       //if(failed)
       //THROW_EXCEPTION(FieldValidationException("Invalid Value", fldName));
       return true;
    }
};
REGISTER_CLASS(${NAME});